src/components/QuoteCard/index.tsx

```diff
@@ -1,89 +1,63 @@
-import { Button } from "../Button";
+import { Quote } from "../..//types";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faThumbsUp as faThumbsUpSolid } from "@fortawesome/free-solid-svg-icons";
+import { faThumbsUp as faThumbsUpRegular } from "@fortawesome/free-regular-svg-icons";
+import { faThumbsDown as faThumbsDownRegular } from "@fortawesome/free-regular-svg-icons";
 import {
-  useQuotesContext,
-  useQuotesDispatchContext,
+  useLikedQuotesContext,
+  useLikedQuotesDispatchContext,
 } from "../../QuotesContextProvider";
-import {
-  useQuoteIndexContext,
-  useQuoteIndexDispatchContext,
-} from "../../QuoteIndexContextProvider";
-import { useState } from "react";
-import { Quote } from "../../types";
+import { useAuth } from "../../AuthContextProvider";
 
-export const QuoteCard = ({ quote, author, likeCount } : Quote) => {
-  const quotes = useQuotesContext();
-  const currentIndex = useQuoteIndexContext();
-  const setQuotes = useQuotesDispatchContext();
-  const dispatchQuoteIndex = useQuoteIndexDispatchContext() || ((index: number) => {});
+interface QuoteCardProps extends Quote {
+  onProfilePage?: boolean;
+  isUserCreated?: boolean;
+  handleDeleteCreatedQuote?: (quote: Quote) => void;
+}
 
-  const [liked, setLiked] = useState(false);
-  const [isBouncing, setIsBouncing] = useState(false);
+export const QuoteCard = ({ quote, author, onProfilePage, isUserCreated, handleDeleteCreatedQuote }: QuoteCardProps) => {
+  const likedQuotes = useLikedQuotesContext() ?? [];
+  const { handleLike, handleDislike } = useLikedQuotesDispatchContext() ?? { handleLike: () => {}, handleDislike: () => {} };
+  const isLiked = likedQuotes.some((q) => q.quote === quote);
+  const { isAuthenticated } = useAuth();
 
-  function handleNextQuoteClick() {
-    if (!quotes) return;
-    const randomIndex = Math.floor(Math.random() * quotes.length);
-    dispatchQuoteIndex && dispatchQuoteIndex(randomIndex);
-    setLiked(false);
-  }
+  return (
+    <section className="bg-white p-6 sm:p-8 rounded-xl max-w-sm sm:max-w-lg md:max-w-xl mx-auto shadow-md border border-gray-200 mt-8 sm:mt-10">
+      <p className="text-xl sm:text-2xl font-bold text-left mb-2 text-gray-800">{quote}</p>
+      <p className="text-sm sm:text-md text-gray-600 text-left mb-4">{author}</p>
 
-  function updateLikeCount() {
-    if (liked) return; // prevent multiple likes
-    setLiked(true);
-    setIsBouncing(true);
+      <div className="flex items-center gap-2 mt-4">
+        {!onProfilePage && (
+          <button
+            className={`p-2 sm:p-3 rounded-full border border-blue-700 text-blue-700 bg-blue-50 transition flex items-center justify-center
+              ${!isAuthenticated ? 'opacity-50 cursor-not-allowed' : 'hover:bg-blue-100'}
+            `}
+            onClick={() => isAuthenticated && handleLike({ quote, author, likeCount: 0 }, !isLiked)}
```

Comment:
If you add disabled attribute as suggested above, you can change this line to:
```suggestion
            onClick={() => handleLike({ quote, author, likeCount: 0 }, !isLiked)}
```

Reply:
I see the disabled is added already, you can just accept the suggestion.